### PR TITLE
PHPC-2319: Bump to libmongoc 1.25.1

### PR DESCRIPTION
**Pull request by alcaeus**
https://jira.mongodb.org/browse/PHPC-2319

This PR also enables builds against the lowest supported (1.25.1) and upcoming stable (r1.25) versions of libmongoc, which was missed in #1486.

### CODE REVIEW

File: .evergreen/config/test-variants.yml [hunk 1]
```diff
@@ -97,31 +97,30 @@ buildvariants:
...
+    tags: ["test", "libmongoc", "debian", "x64"]
+    run_on: debian11-small
+    expansions:
+      FETCH_BUILD_VARIANT: "build-debian11"
+      FETCH_BUILD_TASK: "build-php-8.2-libmongoc-next-stable"
+    depends_on:
+      - variant: "build-debian11"
+        name: "build-php-8.2-libmongoc-next-stable"
+    tasks:
+      - ".replicaset .local .7.0 .auth"
```
Comment (jmikola) [inline .evergreen/config/test-variants.yml:123]:
```suggestion
      - ".replicaset .local .7.0 .auth"

```

File: .evergreen/config/test-variants.yml [hunk 2]
```diff
@@ -97,31 +97,30 @@ buildvariants:
...
+    run_on: debian11-small
+    expansions:
+      FETCH_BUILD_VARIANT: "build-debian11"
+      FETCH_BUILD_TASK: "build-php-8.2-libmongoc-lowest-supported"
+    depends_on:
+      - variant: "build-debian11"
+        name: "build-php-8.2-libmongoc-lowest-supported"
+    tasks:
+      - ".replicaset .local .7.0 .auth"
+  - name: test-debian11-php82-local-libmongoc-next-stable
```
Comment (jmikola) [inline .evergreen/config/test-variants.yml:112]:
```suggestion

  - name: test-debian11-php82-local-libmongoc-next-stable
```

**Review by jmikola — APPROVED** [review]
Some whitespace suggestions, but LGTM.

Note that I also fixed the issue typo in the PR title.

**Comment by alcaeus** [comment]
> Note that I also fixed the issue typo in the PR title

🤦‍♂️ Thank you!

**Pull request merged**